actors/migration/top.go

```diff
@@ -164,17 +173,17 @@ func MigrateStateTree(ctx context.Context, store cbor.IpldStore, stateRootIn cid
 		}
 
 		// Hand off migration of one actor, blocking if we are out of worker goroutines
-		if err := sem.Acquire(ctx, 1); err != nil {
+		if err := syncCtx.sem.Acquire(ctx, 1); err != nil {
 			return err
 		}
-		go migrateOneActor(ctx, store, addr, *actorIn, actorsOut, priorEpoch, powerUpdates, transferCh, errCh)
+		go migrateOneActor(ctx, store, addr, *actorIn, actorsOut, priorEpoch, powerUpdates, syncCtx, transferCh, errCh)
```

Comment:
If you buffer the channels (buffer size = number of workers), you should be able to manage the semaphore entirely in this function.

```suggestion
		go func() {
		    migrateOneActor(ctx, store, addr, *actorIn, actorsOut, priorEpoch, powerUpdates, syncCtx, transferCh, errCh)
		    sem.Release(ctx, 1)
		}()
```